integration_tests/src/main/python/asserts.py

```diff
@@ -638,3 +638,24 @@ def do_it_all(spark):
         else:
             return spark.sql(sql)
     assert_gpu_and_cpu_are_equal_collect(do_it_all, conf, is_cpu_first=is_cpu_first)
+
+def assert_cpu_and_gpu_write_contains_ansi_cast(func, conf):
```

Comment:
does not look this function is write-specific, should it be named `assert_cpu_and_gpu_equals_contains_ansi_cast`

looks like we could remove some boiler plate code by reusing `run_with_cpu_and_gpu`